src/jenkins.ts

```diff
@@ -0,0 +1,202 @@
+   */
+  readonly downloadUrl?: string;
+
+  /**
+   * The name of Jenkins plugin.
+   */
+  readonly name: string;
+
+  /**
+   * The version of Jenkins plugin.
+   */
+  readonly version: string;
+}
+
+export interface SeedJobs {
+  /**
+   * The unique name for the seed job.
+   */
+  readonly id: string;
+
+  /**
+   * The description of the seed job.
+   */
+  readonly description: string;
+
+  /**
+   * The repository branch where seed job definitions are present.
+   */
+  readonly repositoryBranch: string;
+
+  /**
+   * The repository access URL. Supports SSH and HTTPS.
+   */
+  readonly repositoryUrl: string;
+
+  /**
+   * The repository path where seed job definitions are present.
+   */
+  readonly targets: string;
+}
+
+export interface JenkinsProps {
+  readonly namespace?: string;
+  readonly labels?: {[name: string]: string};
+  readonly disableCsrfProtection?: boolean;
+  readonly basePlugins?: Plugins[];
+  readonly plugins?: Plugins[];
+  readonly seedJobs?: SeedJobs[];
+}
+
+export class Jenkins extends Construct {
+  constructor(scope: Construct, id: string, props: JenkinsProps = {}) {
+    super(scope, id);
+
+    const namespace = props.namespace ?? 'default';
+    const labels = props.labels ?? { app: 'jenkins' };
+    const authorizationStrategy = 'createUser';
+    const disableCSRFProtection = props.disableCsrfProtection ?? false;
+    const basePlugins = getJenkinsBasePlugins(props.basePlugins);
+    const plugins = props.plugins ?? [];
+    const containers = getContainerConfiguration();
+    const seedJobs = props.seedJobs ?? [];
+
+    const jenkinsCustomResource = new JenkinsCustomResource(this, 'JenkinsCustomResource', {
+      metadata: {
+        namespace: namespace,
+        labels: labels,
+        name: 'cdk8s-jenkins-construct',
+      },
+      spec: {
+        groovyScripts: {
+          configurations: [],
+          secret: {
+            name: '',
+          },
+        },
+        configurationAsCode: {
+          configurations: [],
+          secret: {
+            name: '',
+          },
+        },
+        jenkinsApiSettings: {
+          authorizationStrategy: authorizationStrategy,
+        },
+        master: {
+          disableCsrfProtection: disableCSRFProtection,
+          containers: containers,
+          basePlugins: basePlugins,
+          plugins: plugins,
+        },
+        seedJobs: seedJobs,
+      },
+    });
+
+    this.node.defaultChild = jenkinsCustomResource;
+  }
+
+  public addBasePlugins(basePluginsUpdates: Plugins[]): void {
+    const jenkinsApiObject = ApiObject.of(this);
+
+    basePluginsUpdates.forEach((plugin) => {
+      jenkinsApiObject.addJsonPatch(JsonPatch.add('/spec/master/basePlugins/-', plugin));
+    });
+  }
+
+  public addPlugins(plugins: Plugins[]): void {
+    const jenkinsApiObject = ApiObject.of(this);
+
+    plugins.forEach((plugin) => {
+      jenkinsApiObject.addJsonPatch(JsonPatch.add('/spec/master/plugins/-', plugin));
+    });
```

Comment:
JSON patches are primarily meant to be a mechanism for userland code (where you can't modify the "internals" of a construct / abstraction). Usually we try to avoid them when we're creating our own new constructs/abstractions from scratch.

Instead, let's add a private variable to this construct that stores any `Plugin`s, and pass that private variable to the underlying function. For example:

```ts
const jenkinsCustomResource = new JenkinsCustomResource(this, 'JenkinsCustomResource', {
  // ...
  master: {
    // ...
    plugins: this._plugins
  }
}
```
This should ensure the entire list is included in the synthesized template.

(If you need to include a more complex value than array, you can accomplish something similar by including a "Lazy" value, like this):
```ts
    plugins: Lazy.any({ produce: () => this.renderPlugins() })
```

Reply:
Ahh, I see. Let me try doing this. This is much better. 
I was not sure if we utilize JsonPatches while making a library. It was looking like a hacky way to accomplish adding to a resource. Thank you for letting me know. :)